### PR TITLE
Chore: CSS cleanup + dark mode support 

### PR DESCRIPTION
**Pull request by znicholasbrown**
For real this time; requires https://github.com/PrefectHQ/prefect-design/pull/600

Note: this will fail build until the next prefect-design release

### CODE REVIEW

File: demo/components/PContextAccordionItem.vue [hunk 1]
```diff
@@ -1,153 +0,0 @@
-<template>
-  <div class="p-context-accordion-item">
```
Comment (pleek91) [inline demo/components/PContextAccordionItem.vue:2]: Why was this component deleted?

File: demo/components/ResizableSection.vue [hunk 2]
```diff
@@ -21,32 +21,46 @@
...
+
+
   const classes = computed(() => ({
+    content: {
+      [getColorModeClass(activeColorMode.value)]: true,
+      'dark': colorTheme.value === 'dark',
+      'light': colorTheme.value === 'light',
+      'bg-background-700': colorTheme.value === 'light',
+      'bg-background-400': colorTheme.value === 'dark',
+    },
```
Comment (pleek91) [inline demo/components/ResizableSection.vue:49]:
```suggestion
    content: [getColorModeClass(activeColorMode.value, {
      'dark': colorTheme.value === 'dark',
      'light': colorTheme.value === 'light',
      'bg-background-700': colorTheme.value === 'light',
      'bg-background-400': colorTheme.value === 'dark',
    }],
```

File: demo/utilities/colorMode.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { useLocalStorage } from '@prefecthq/vue-compositions'
```
Comment (pleek91) [inline demo/utilities/colorMode.ts:1]: This file would make a nice composition. Don't bother but would be cool at some point.

File: src/components/CodeSnippet.vue [hunk 4]
```diff
@@ -38,7 +38,6 @@
 
 <style>
 .code-snippet { @apply
-  bg-slate-700
```
Comment (pleek91) [inline src/components/CodeSnippet.vue:41]: does a code snippet really not need a background?

File: src/components/FlowRunCreateForm.vue [hunk 5]
```diff
@@ -23,13 +23,7 @@
...
 
       <p-label label="Tags (optional)">
-        <p-tags-input v-model="tags" :options="deploymentTags">
-          <template #default="{ selectedOption }">
-            <p-tag :dismissible="!selectedOption.disabled">
-              {{ selectedOption.label }}
-            </p-tag>
-          </template>
-        </p-tags-input>
+        <p-tags-input v-model="tags" :options="deploymentTags" />
```
Comment (pleek91) [inline src/components/FlowRunCreateForm.vue:26]: Ah is this what the update to `p-tags-input` was for?

File: src/components/FlowsTable.vue [hunk 6]
```diff
@@ -197,7 +188,7 @@
 }
 
 .flows-table__activity-chart { @apply
-  h-12
+  !h-12
```
Comment (pleek91) [inline src/components/FlowsTable.vue:191]: Seems suspicious. Why the `!`?

File: src/components/FourZeroFour.vue [hunk 7]
```diff
@@ -1,38 +1,35 @@
...
....1-3.4v-6.8h-2.6v-2h2.6v-4.4h2.2v4.4h4.8v2h-4.8v6.8c0 .9.2 1.5.6 2 .4.3 1 .5 1.7.5l1-.1a8 8 0 0 0 1.2-.3l.5 1.9a8 8 0 0 1-2.9.5Zm20.1-.2v-17.5H351v2h-9.2v6.8h7.6v2h-7.6v6.7h-2.2Zm21 .2c-1.7 0-3-.4-4-1.2-1-.8-1.5-2-1.5-3.7v-3.6c0-1.7.5-3 1.5-3.8a6 6 0 0 1 4-1.2 6 6 0 0 1 3.9 1.2c1 .9 1.5 2.1 1.5 3.8v3.6c0 1.7-.5 2.9-1.5 3.7-1 .8-2.3 1.2-4 1.2Zm0-2c1 0 1.9-.2 2.4-.6.5-.5.8-1.3.8-2.3v-3.6c0-1-.3-1.8-.8-2.3-.5-.5-1.4-.7-2.5-.7-1 0-1.9.2-2.4.7-.6.5-.8 1.3-.8 2.3v3.6c0 1 .2 1.8.8 2.3.5.4 1.4.7 2.4.7Zm14.2 2a4 4 0 0 1-3-1.2 4.5 4.5 0 0 1-1.2-3.3V492h2.2v8.4c0 1 .2 1.6.6 2.1.4.5 1 .7 2 .7a4 4 0 0 0 2-.5c.6-.4 1.2-1 1.9-1.7v-9h2.2V505h-2.2v-2.2h-.7c-.3.7-.8 1.3-1.5 1.8-.7.4-1.4.6-2.3.6Zm11.3-.2v-13h2.1v2h.7c.3-.6.8-1.2 1.5-1.7.7-.4 1.5-.6 2.3-.6a4 4 0 0 1 3.1 1.2c.8.8 1.2 1.9 1.2 3.3v8.8h-2.2v-8.4c0-1-.3-1.7-.8-2.1-.4-.5-1-.7-1.8-.7s-1.6.2-2.2.6c-.6.4-1.2 1-1.8 1.7v8.9H386Zm20.2-1.9c.8 0 1.5-.1 2-.5.7-.4 1.3-1 2-1.7v-6.5l-1.5-.6a6 6 0 0 0-1.6-.2c-1.1 0-2 .3-2.6.8-.6.5-.9 1.3-.9 2.3v3.5c0 1.2.3 2 .7 2.4a3 3 0 0 0 2 .5Zm3.3-.3c-.4.8-.9 1.4-1.5 1.8-.7.4-1.4.6-2.3.6a4 4 0 0 1-3.1-1.3 5.4 5.4 0 0 1-1.2-3.7v-3.5c0-1.5.5-2.8 1.5-3.7a6 6 0 0 1 4.2-1.3 7.6 7.6 0 0 1 3 .6v-4.8h2.2V505h-2.1v-2.2h-.7Z"
```
Comment (pleek91) [inline src/components/FourZeroFour.vue:11]: lol what was this path? Something you cannot even see?

File: src/components/FourZeroFour.vue [hunk 8]
```diff
@@ -332,32 +329,43 @@
 </script>
 
 <style>
-.not-found {
+.four-zero-four {
   display: flex;
   align-items: center;
   justify-content: center;
-  background:#0a2053;
+}
```
Comment (pleek91) [inline src/components/FourZeroFour.vue:336]: This could be tailwind and combined with the next selector right?

File: src/components/JsonInput.vue [hunk 9]
```diff
@@ -90,7 +90,7 @@
   w-full
   h-full
   min-h-[inherit]
-  bg-slate-700
+  bg-slate-800 /* This should match the background color of the base input component in its darkest mode */
```
Comment (pleek91) [inline src/components/JsonInput.vue:93]: Is this needed because the JsonInput is basically always in dark mode?

File: src/components/LogsContainer.vue [hunk 10]
```diff
@@ -62,7 +62,7 @@
   flex
   justify-center
   w-full
-  bg-slate-700
+  bg-slate-800
```
Comment (pleek91) [inline src/components/LogsContainer.vue:65]: Should the logs container and the JsonInput components have a "lite" version so that way they can just inherit the default light and dark mode colors? Rather than always being "dark"

File: src/components/RadarNodePlaceholder.vue [hunk 11]
```diff
@@ -30,17 +44,18 @@
...
 
 .radar-node-placeholder::after {
   animation: node-placeholder-animation 2s cubic-bezier(0, 0.35, 1, 0.35) infinite;
   background: linear-gradient(
     90deg,
-    rgba(255, 255, 255, 0) 0%,
-    rgba(255, 255, 255, 0.2) 20%,
-    rgba(255, 255, 255, 0.5) 60%,
-    rgba(255, 255, 255, 0.1)
+    hsla(v-bind('style.placeholder'), 0) 0%,
```
Comment (pleek91) [inline src/components/RadarNodePlaceholder.vue:54]: the fact that this works is amazing

File: src/components/TimezoneSelect.vue [hunk 12]
```diff
@@ -14,6 +14,7 @@
 
   const props = defineProps<{
     modelValue: string | null,
+    showTimestamp?: boolean,
```
Comment (pleek91) [inline src/components/TimezoneSelect.vue:17]: seems like a pretty unrelated change. Was this a bug you stumbled across?

File: src/components/WorkPoolQueueEditForm.vue [hunk 13]
```diff
@@ -107,7 +107,7 @@
 <style>
 .work-pool-queue-edit-form { @apply
   border
-  border-gray-300
+  dark:border-background-600
```
Comment (pleek91) [inline src/components/WorkPoolQueueEditForm.vue:110]: if this was wrapped in a card would this just work?

File: tsconfig.json [hunk 14]
```diff
@@ -28,7 +28,11 @@
     "skipLibCheck": true,
     "sourceMap": true,
     "strict": true,
-    "target": "esnext"
+    "target": "esnext",
+    "types": [
+      "vite-svg-loader",
+      "vite/client"
```
Comment (pleek91) [inline tsconfig.json:34]: what did adding these fix?

File: demo/components/PContextAccordionItem.vue (hunk 1 above)
Comment (znicholasbrown) [reply demo/components/PContextAccordionItem.vue:2]: It's no longer needed since the fix from https://github.com/PrefectHQ/prefect-design/pull/588 went live; we can use the `PContextAccordionItem` component from `prefect-design` instead of duplicating it now.

File: src/components/CodeSnippet.vue (hunk 4 above)
Comment (znicholasbrown) [reply src/components/CodeSnippet.vue:41]: This is a wrapper for lang-specific code views; those have their own backgrounds already for syntax highlight contrast so this was a duplicate (and subtly incorrect) bg color.

File: demo/components/PContextAccordionItem.vue (hunk 1 above)
Comment (pleek91) [reply demo/components/PContextAccordionItem.vue:2]: ah nice, forgot it was duplicated.

File: src/components/FlowRunCreateForm.vue (hunk 5 above)
Comment (znicholasbrown) [reply src/components/FlowRunCreateForm.vue:26]: Yeah exactly haha

File: src/components/FlowsTable.vue (hunk 6 above)
Comment (znicholasbrown) [reply src/components/FlowsTable.vue:191]: Honestly might not be needed, the chart component doesn't do a good job growing to allowed space so this was a way to force it to do so. Maybe higher CSS specificity is needed?

File: src/components/FourZeroFour.vue (hunk 7 above)
Comment (znicholasbrown) [reply src/components/FourZeroFour.vue:11]: Nah, this was the text glyphs converted to a single path, which is why this component never responded to global font changes

File: src/components/FlowsTable.vue (hunk 6 above)
Comment (pleek91) [reply src/components/FlowsTable.vue:191]: Not a big deal. But if there's a specificity issue that would be something to fix possibly. We shouldn't have to worry about specificity especially for something like height.

File: src/components/FourZeroFour.vue (hunk 7 above)
Comment (pleek91) [reply src/components/FourZeroFour.vue:11]: 😂 nice

File: src/components/FourZeroFour.vue (hunk 8 above)
Comment (znicholasbrown) [reply src/components/FourZeroFour.vue:336]: Yeah I think so? I'm not entirely sure why these were written this way, it's possible there's an SVG style scoping issue? I'd rather not dig into it in this PR tbh

File: src/components/JsonInput.vue (hunk 9 above)
Comment (znicholasbrown) [reply src/components/JsonInput.vue:93]: Correct - all our custom code snippets are dark themed already, this just brings the styling in line with other components (I've added this comment anywhere I needed to add an explicit bg color)

File: src/components/LogsContainer.vue (hunk 10 above)
Comment (znicholasbrown) [reply src/components/LogsContainer.vue:65]: I thought about that but since both of those are pretty heavy on the custom styling I figure if there's enough demand we can tackle those later. Personally I kinda like the visual contrast in the light theme

File: src/components/RadarNodePlaceholder.vue (hunk 11 above)
Comment (znicholasbrown) [reply src/components/RadarNodePlaceholder.vue:54]: I KNOW. It feels like magic!!

File: src/components/TimezoneSelect.vue (hunk 12 above)
Comment (znicholasbrown) [reply src/components/TimezoneSelect.vue:17]: Ah no, it looks like git didn't do a great job fast forwarding this branch on [this commit](https://github.com/PrefectHQ/orion-design/commit/1773abab0ab1431dfcaabc1be3e5f39bc10ed57c)

Comment (znicholasbrown) [reply src/components/TimezoneSelect.vue:17]: I think it's probably ok though because @stackoverfloweth is still listed as the co-author so it'll just get fast forwarded when we merge

File: src/components/LogsContainer.vue (hunk 10 above)
Comment (pleek91) [reply src/components/LogsContainer.vue:65]: I do to but only because I want everything dark lol. It would just be nice if these didn't have special styling.

File: src/components/WorkPoolQueueEditForm.vue (hunk 13 above)
Comment (znicholasbrown) [reply src/components/WorkPoolQueueEditForm.vue:110]: Yeah I think there's a small-ish ticket to create a common interface for full-page forms

File: tsconfig.json (hunk 14 above)
Comment (znicholasbrown) [reply tsconfig.json:34]: Just TS being big mad when working in the vite config and refusing to do any type validation and also allows us to use `import.meta.hot` without adding it to our own meta definition

File: src/components/FlowsTable.vue (hunk 6 above)
Comment (znicholasbrown) [reply src/components/FlowsTable.vue:191]: Agreed _usually_; these canvas and svg charts are a little weird when it comes to responsiveness

File: demo/utilities/colorMode.ts (hunk 3 above)
Comment (znicholasbrown) [reply demo/utilities/colorMode.ts:1]: Agreed, I think I've created this or something like it in a few places now

File: demo/components/ResizableSection.vue (hunk 2 above)
Comment (znicholasbrown) [reply demo/components/ResizableSection.vue:49]: Ah actually `p-frame` is lazily looking for `Record<string, unknown>` as a `bodyClass` type which means we can't use this syntax

Comment (pleek91) [reply demo/components/ResizableSection.vue:49]: 🤔  these should be the same?  
```
{ 
  [getColorModeClass(activeColorMode.value)]: true
}
```
```
[getColorModeClass(activeColorMode.value]
```

Comment (pleek91) [reply demo/components/ResizableSection.vue:49]: oh I get your comment now. Looking at that

Comment (znicholasbrown) [reply demo/components/ResizableSection.vue:49]: Yeah it was probably cause I wrote the prop that way tbh but I think there's a good reason to; passing around the vue class and style types is really slow for TS cause they're huge

Comment (pleek91) [reply demo/components/ResizableSection.vue:49]: Vue has some helper types for this sort of this. Anyway, carry on. I'll create an issue in prefect-design to support vue style dynamic classes

Comment (pleek91) [reply demo/components/ResizableSection.vue:49]: https://github.com/PrefectHQ/prefect-design/issues/603

**Review by pleek91 — APPROVED** [review]
